streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskManager.java

```diff
@@ -220,12 +215,18 @@ public void handleAssignment(final Map<TaskId, Set<TopicPartition>> activeTasks,
             } else {
                 try {
                     task.suspend();
-                    final Map<TopicPartition, OffsetAndMetadata> committableOffsets = task.prepareCommit();
-
-                    tasksToClose.add(task);
-                    if (!committableOffsets.isEmpty()) {
-                        consumedOffsetsAndMetadataPerTask.put(task.id(), committableOffsets);
+                    if (task.commitNeeded()) {
```

Comment:
nit: add a comment above `task.suspend()` that for active it should always be an no-op?

Reply:
ack

Reply:
Actually it's not _always_ a no-op, since we will bail on suspending/committing the remaining active tasks if any of them throws an exception in `handleRevocation` -- so it's possible some active tasks are not yet suspended at this point.
I will add a comment to clarify

Reply:
Not sure if I can follow -- if it's a no-op, why do we call it? Or do you say, we need to tall if for standbies as we don't suspend them presiously?

Reply:
We need to do it for all standbys, and we may need to do it for some actives. Since suspending is now idempotent we may as well just call `suspend` universally

Reply:
For this case, it won't be a no-op for some active tasks? So we should not have such a comment?

Reply:
The current comment is
```
// Should be a no-op for active tasks, unless we hit an exception during handleRevocation
```